### ISSUE TITLE
After 78 iterations, the loss returned nan

### ISSUE THREAD

**Issue by igorkf** [jrzaurin/LightGBM-with-Focal-Loss#5]
Hi. Thanks for this implementation.
I had a problem:
```
/var/folders/tx/5cvg8b910f31k6ttxp5ds9cxbmnldx/T/ipykernel_14194/2624284624.py:26: RuntimeWarning: overflow encountered in exp
  p = 1/(1+np.exp(-x))
/var/folders/tx/5cvg8b910f31k6ttxp5ds9cxbmnldx/T/ipykernel_14194/2624284624.py:27: RuntimeWarning: divide by zero encountered in log
  return -( a*t + (1-a)*(1-t) ) * (( 1 - ( t*p + (1-t)*(1-p)) )**g) * ( t*np.log(p)+(1-t)*np.log(1-p) )
/var/folders/tx/5cvg8b910f31k6ttxp5ds9cxbmnldx/T/ipykernel_14194/2624284624.py:27: RuntimeWarning: invalid value encountered in multiply
  return -( a*t + (1-a)*(1-t) ) * (( 1 - ( t*p + (1-t)*(1-p)) )**g) * ( t*np.log(p)+(1-t)*np.log(1-p) )
 ```

Do you know what could it be?
The problem is a multiclass classification with 30 classes.

**Comment by jrzaurin** [comment]
Hey sorry for the late reply, been on holidays. Give me a few days and Ill have a look  but normally these things dissapear when playing with the learning rate

**Comment by igorkf** [comment]
No problem! I just though would be helpful to tell you.
